android/quest/src/test/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModelTest.kt

```diff
@@ -1144,4 +1146,73 @@ class QuestionnaireViewModelTest : RobolectricTest() {
       bundle.entry.any { it.resource is Basic && it.resource.id == "basic-resource-id" },
     )
   }
+
+  @Test
+  fun testSaveExtractedResourcesAddsRelatedEntityLocationMetaTagToExtractedResource() = runBlocking {
+    val bundleSlot = slot<Bundle>()
+    val bundle = bundleSlot.captured
+    val linkId = "linkId"
+    val questionnaire = extractionQuestionnaire()
+    val questionnaireConfig =
+      questionnaireConfig.copy(
+        resourceIdentifier = "resourceId",
+        resourceType = ResourceType.Location,
+        saveQuestionnaireResponse = false,
+        type = "EDIT",
+        linkIds = listOf(LinkIdConfig(linkId = linkId, LinkIdType.LOCATION)),
+        extractedResourceUniquePropertyExpressions =
+          listOf(
+            ExtractedResourceUniquePropertyExpression(
+              ResourceType.Location,
+              "Observation.code.where(coding.code='obs1').coding.code",
+            ),
+          ),
+      )
+    val previousObs =
+      Observation().apply {
+        id = "previousObs1"
+        code = (CodeableConcept(Coding("http://obsys", "obs1", "Obs 1")))
+      }
+
+    val questionnaireResponse =
+      extractionQuestionnaireResponse().apply {
+        val extractionDate = Date()
+        subject = patient.asReference()
+        val listResource =
+          ListResource().apply {
+            id = UUID.randomUUID().toString()
```

Comment:
Use a dummy test uuid so that you can use it in the assert as well (to confirm it is actually the correct one added)